src/iexamine.cpp

```diff
@@ -630,22 +630,65 @@ void iexamine::cardreader(player *p, map *m, int examx, int examy)
 void iexamine::rubble(player *p, map *m, int examx, int examy)
 {
     bool has_digging_tool = p->has_items_with_quality( "DIG", 2, 1 );
-  // Perhaps check for vehicle covering the rubble and bail out if so (string freeze ATM)?
+    // Perhaps check for vehicle covering the rubble and bail out if so (string freeze ATM)?
     if (!has_digging_tool) {
         add_msg(m_info, _("If only you had a shovel..."));
         return;
     }
+
+    // Ask if there's something possibly more interesting than this rubble here
     std::string xname = m->furnname(examx, examy);
-    if (query_yn(_("Clear up that %s?"), xname.c_str())) {
-        // "Remove"
-        p->moves -= 200;
-        m->furn_set(examx, examy, f_null);
+    if( ( m->veh_at( examx, examy ) != nullptr ||
+          m->tr_at( examx, examy ) != tr_null ||
+          g->critter_at( examx, examy ) != nullptr ) &&
+          !query_yn(_("Clear up that %s?"), m->tername(examx, examy).c_str() ) ) {
```

Comment:
Previously `xname` (which is `furnname`), now it's `tername` and the function actually removes the furniture.